tests/buffer_ut/buffer_ut.c

```diff
@@ -1359,6 +1362,29 @@ TEST_FUNCTION(BUFFER_prepend_Succeed)
         ASSERT_IS_NULL(result);
     }
 
+    /* Tests_SRS_BUFFER_88_003: [ If any error occurs, BUFFER_clone shall return NULL. ] */
+    TEST_FUNCTION(BUFFER_Clone_malloc_fails)
+    {
+        ///arrange
+        BUFFER_HANDLE g_hBuffer;
+        g_hBuffer = BUFFER_new();
+        (void)BUFFER_build(g_hBuffer, BUFFER_TEST_VALUE, ALLOCATION_SIZE);

```

Comment:
>BUFFER_build [](http://example.com/codeflow?start=14&length=12)

This should also be asserted that it succeeds when it is part of the test setup. Here and in the other tests with the same pattern of using BUFFER_build in test setup.